src/services/sync/strategies/consumerActions/UserAction.js

```diff
@@ -0,0 +1,63 @@
+const _ = require('lodash');
+
+const BaseConsumerAction = require('./BaseConsumerAction');
+// TODO: place from where it is importat must be fixed later
+const { LDAP_SYNC_ACTIONS } = require('../LDAPSyncer');
+const { SchoolRepo, UserRepo } = require('../../repo');
+
+const defaultOptions = {
+	allowedLogKeys: ['ldapId', 'systemId', 'roles', 'activated'],
+};
+
+class UserAction extends BaseConsumerAction {
+	constructor(filterActive = true, options = defaultOptions) {
+		super(LDAP_SYNC_ACTIONS.SYNC_USER, options);
+		this.filterActive = filterActive;
+	}
+
+	async action(data = {}) {
+		const { user = {}, account = {} } = data;
```

Comment:
where are params validated? why not use named params in the action?

Reply:
Data come from the queue. To every actions it exist one queue key. In later iterations it make sense to bring it near to the action class and share the interface. But not in this iteration.

Reply:
hm, but how do i know what the data looks like? that should be anywhere close to here to know it i mean...